Make EXIF matching default and add optional filename matching

NightMean opened the pull request:
This commit updates the MotionPhoto2 script to use EXIF-based matching (`MakerNotes:ContentIdentifier` for images and `QuickTime:ContentIdentifier` for videos) as the default method for pairing image and video files in directory processing, addressing issue #20. 
Filename-based matching is retained as an optional feature (`--use-filename-matching`) under the "Settings" group in the GUI

Key changes:
- Added `--use-filename-matching` flag to toggle between EXIF (default) and filename matching.
- Fixed EXIF matching logic to correctly pair files by stripping whitespace from `ContentIdentifier` values and adding verbose debug output.
- Preserved original filename matching logic, with improved feedback when no match is found.

Resolves #20.
Made with Grok 3 (Think)

While I'm not a programmer and I only have basic understanding, In this modern age with AI, I took a look at this issue with [Grok](https://grok.com/). 
With just 2 simple prompts, I was able to make a fully working prototype. I've asked it to use the EXIF matching as default with an optional filename matching. 
From my testing, it works exactly as I needed it to. I'm not sure if this is how you imagined this feature but feel free to modify it as you wish :)

NightMean commented:
While I'm not a programmer and I only have basic understanding, In this modern age with AI, I took a look at this issue with [Grok](https://grok.com/). 
With just 2 simple prompts, I was able to make a fully working prototype. I've asked it to use the EXIF matching as default with an optional filename matching. 
From my testing, it works exactly as I needed it to. I'm not sure if this is how you imagined this feature but feel free to modify it as you wish :)

PetrVys commented:
Thanks, that looks very good for an AI result, actually. I'll try to review it soon, but cursory reading of the patch looks ok and do exactly what's needed.

PetrVys commented:
BTW - on top of cosmetic changes, I've reverted the default back to filename based matching _if run from commandline_. I don't want to break any scripts someone might've created, although I agree it is the more useful mode :-).
When run in GUI mode, using exif match is the new default.

NightMean commented:
> Thanks, that looks very good for an AI result, actually. I'll try to review it soon, but cursory reading of the patch looks ok and do exactly what's needed.

Thanks! That's exactly what I was trying to achieve :) 

> BTW - on top of cosmetic changes, I've reverted the default back to filename based matching if run from commandline. I don't want to break any scripts someone might've created, although I agree it is the more useful mode :-).
When run in GUI mode, using exif match is the new default.

That's understandable, thanks for looking into it so quickly. 

By the way, I modified the Readme under Notes as well with the addition of the `--filename-match` which is still present. 
As you renamed it to `--exif-match` and changed the behavioru, that sentence should be adjusted.

PetrVys commented:
Thanks, updated.
